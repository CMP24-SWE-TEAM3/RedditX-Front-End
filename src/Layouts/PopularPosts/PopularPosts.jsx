// imports
import React from 'react';
import HotButton from "Components/HotButton/HotButton";
import {Container} from "./PopularPosts.styled";
import EveryWhereButton from "Components/EveryWhereButton/EveryWhereButton";
import NewButton from "Components/NewButton/NewButton";
import TopButton from "Components/TopButton/Top";
import DotButton from "Components/DotButton/DotButton";
import {HiFire} from "react-icons/hi";
import {RiRocketLine} from 'react-icons/ri';

/**
 * Component that displays a list of posts from a category list view
 * @returns {React.Component}
 */
const PopularPost = () => {
    return (
        <Container>
            <HotButton ico={<RiRocketLine/>} name={'Best'}/>
            <HotButton ico={<HiFire/>} name={'Hot'} />
            {/*<EveryWhereButton/>*/}
            <NewButton/>
            <TopButton/>
            <DotButton/>
            {/*<CategoryButton/>*/}
        </Container>
    );
};

export default PopularPost;
