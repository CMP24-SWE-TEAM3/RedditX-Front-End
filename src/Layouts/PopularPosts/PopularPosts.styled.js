import styled from "styled-components";

export const Container = styled.div`
  display: flex;
  height: 58px;
  //left: auto;
  border-radius: 4px;
  //position: relative;
  background-color: ${({theme})=>theme.background.primary};
  @media only screen and (max-width: 961px) {
    width: 100%;
  }
  align-items: center;
  border: 1px solid ${({theme}) => theme.background.primary};
  box-sizing: border-box;
  flex-flow: row nowrap;
  justify-content: flex-start;
  margin-bottom: 16px;
  padding: 10px 12px;
`