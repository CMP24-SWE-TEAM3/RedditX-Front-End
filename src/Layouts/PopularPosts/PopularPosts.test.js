import {shallow} from 'enzyme';

// import Components
import PopularPosts from "./PopularPosts";

describe("Popular Posts Component", () => {
    it("this is a test for Popular Posts Component", () => {
        expect(shallow(<PopularPosts/>)).toMatchSnapshot();

    });
});