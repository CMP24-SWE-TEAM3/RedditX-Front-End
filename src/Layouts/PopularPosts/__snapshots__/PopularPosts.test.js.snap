// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popular Posts Component this is a test for Popular Posts Component 1`] = `ShallowWrapper {}`;
